libandroid-navigation/src/test/java/com/mapbox/services/android/navigation/v5/navigation/MapboxNavigationTest.kt

```diff
@@ -300,6 +350,198 @@ class MapboxNavigationTest : BaseTest() {
+        }
+        assertTrue(tilePath.captured.contains("2019_04_13-00_00_11"))
+    }
+
+    @Test
+    fun enableFreeDrive_freeDriveLocationUpdaterStartIsCalledWhenOnConfigured() {
+        val mockedFreeDriveLocationUpdater = mockk<FreeDriveLocationUpdater>(relaxed = true)
+        val navigation = MapboxNavigation(
+            RuntimeEnvironment.application,
+            ACCESS_TOKEN,
+            mockk<MapboxNavigationOptions>(relaxed = true),
+            mockk<NavigationTelemetry>(relaxed = true),
+            mockk<LocationEngine>(relaxed = true),
+            mockk<Navigator>(relaxed = true),
+            mockedFreeDriveLocationUpdater
+        )
+        val onOfflineTilesConfiguredCallback = slot<OnOfflineTilesConfiguredCallback>()
+
+        navigation.enableFreeDrive()
+
+        verify(exactly = 1) {
+            mockedFreeDriveLocationUpdater.configure(
+                any(),
+                capture(onOfflineTilesConfiguredCallback)
+            )
+        }
+        onOfflineTilesConfiguredCallback.captured.onConfigured(3)
+        verify(exactly = 1) {
+            mockedFreeDriveLocationUpdater.start()
+        }
+    }
+
+    @Test
+    fun enableFreeDrive_freeDriveLocationUpdaterStartIsNotCalledWhenOnConfigurationError() {
+        val mockedFreeDriveLocationUpdater = mockk<FreeDriveLocationUpdater>(relaxed = true)
+        val navigation = MapboxNavigation(
+            RuntimeEnvironment.application,
+            ACCESS_TOKEN,
+            mockk<MapboxNavigationOptions>(relaxed = true),
+            mockk<NavigationTelemetry>(relaxed = true),
+            mockk<LocationEngine>(relaxed = true),
+            mockk<Navigator>(relaxed = true),
+            mockedFreeDriveLocationUpdater
+        )
+        val onOfflineTilesConfiguredCallback = slot<OnOfflineTilesConfiguredCallback>()
+
+        navigation.enableFreeDrive()
+
+        verify(exactly = 1) {
+            mockedFreeDriveLocationUpdater.configure(
+                any(),
+                capture(onOfflineTilesConfiguredCallback)
+            )
+        }
+        onOfflineTilesConfiguredCallback.captured.onConfigurationError(OfflineError("test"))
+        verify(exactly = 0) {
+            mockedFreeDriveLocationUpdater.start()
+        }
+    }
+
+    @Test
+    fun enableFreeDrive_freeDriveLocationUpdaterStartIsCalledRightAwayIfAlreadyConfigured() {
+        val mockedFreeDriveLocationUpdater = mockk<FreeDriveLocationUpdater>(relaxed = true)
+        val navigation = MapboxNavigation(
+            RuntimeEnvironment.application,
+            ACCESS_TOKEN,
+            mockk<MapboxNavigationOptions>(relaxed = true),
+            mockk<NavigationTelemetry>(relaxed = true),
+            mockk<LocationEngine>(relaxed = true),
+            mockk<Navigator>(relaxed = true),
+            mockedFreeDriveLocationUpdater
+        )
+        val onOfflineTilesConfiguredCallback = slot<OnOfflineTilesConfiguredCallback>()
+        navigation.enableFreeDrive()
```

Comment:
We don't need to invoke `enableFreeDrive` here. We should only invoke it after `onOfflineTilesConfiguredCallback` returns and check for on invocation of `mockedFreeDriveLocationUpdater.start()`. Checking for 2 invocations overlaps with one of the tests above already.

Reply:
This test is specifically testing this branch of `enableFreeDrive` https://github.com/mapbox/mapbox-navigation-android/blob/6a934f45a81d44f6d7697c9750c0f3a4027ca4f9/libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/navigation/MapboxNavigation.java#L713-L717 which happens if and only if is configured previously (that's why this `enableFreeDrive` needs to be invoke as part of the Arrange of the test). Does that make sense?

Reply:
Ah yes :+1: